.github/workflows/update-badge.yml

```diff
@@ -0,0 +1,59 @@
+# Note: this workflow requires the repository to give Write access to Github Workflows.
+# in Settings > Actions > General > Workflow permissions
```

Comment:
We can make the permission requirement explicit - https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs. If we specify them at top level, they're applied to all jobs.

```suggestion
# Note: this workflow requires the repository to give Write access to Github Workflows.
# in Settings > Actions > General > Workflow permissions

permissions:
  contents: write
```